app/web/features/search/MapWrapper.tsx

```diff
@@ -160,6 +181,29 @@ export default function MapWrapper({
     [map]
   );
 
+  /**
+   * Checks if filters are applied
+   */
+  useEffect(() => {
+    const filtersActive =
+      queryName !== "" ||
+      lastActiveFilter !== 0 ||
+      hostingStatusFilter.length !== 0 ||
+      numberOfGuestFilter !== undefined ||
+      completeProfileFilter !== false ||
+      (locationResult?.location.lng !== 0 &&
+        locationResult?.location.lat !== 0);
+
+    setAreFiltersApplied(filtersActive);
+  }, [
+    queryName,
+    lastActiveFilter,
+    hostingStatusFilter,
+    numberOfGuestFilter,
+    completeProfileFilter,
+    locationResult,
+  ]);
+
```

Comment:
Okay first of all, this was a great first attempt, and it pretty much works! Well done taking this on as one of your first tickets as honestly the map is confusing even for me!

A few notes on things I'd change due to the approach:

First a caveat - this isn't how I'd ideally do this in i.e. a day job, normally I'd just send an api call to the backend to clear everything and that would return an updated response, which would update the UI.

BUT we cache the result and filter it on the frontend, which is why that doesn't work here.

Given we will refactor the map soon anyway, this is how I'd go about it as a stop gap:

We already have a very similar useEffect to the one you created for `wasSearchPerformed` in the parent component SearchPage so IMO we want to avoid another one.

Also I'd generally try to keep this functionality in the same place the values are set in useState, which is in SearchPage.

Since you are correctly clearing the values there, we just need a variable to pass down to say "hey the filters changed refilter the user pin results in this component. Your function is already updating the state, which is re-running the api call and regenerating "result", which gets passed down to `MapWrapper`.

So IMO we just need to tell MapWrapper the search was cleared as the actual functionality is already there.

I'd simplify it a bit like so:

- In SearchPage create a useState variable `areFiltersCleared, setAreFiltersCleared` and set it to true as a default value
- When you run the handleClearFilters function in SearchPage, set this to true
- In SearchPage, in the useEffect that checks if `if (!wasSearchPerformed)` create an else to the inner if clause that checks if everything is 0 or undefined, and `setAreFiltersCleared` false there
- In MapWrapper, in the useEffect that re-renders users list on map, change `wasSearchedPerformed` to `{wasSearchPerformed || areFilltersCleared)`

I thiiiiink that should do it but let me know if not!

Reply:
Awesome, thank you so much for the detailed feedback and explanation Nicole! Really appreciate it! 

I've made the changes you outlined, however I just want to call out that updating the useEffect that checks if `if (!wasSearchPerformed)` in SearchPage differed a bit from what you proposed... this was to ensure that the areFiltersCleared state was properly updated all the time, not just upon the first search. LMK if this makes more sense now.